sympy/matrices/expressions/matmul.py

```diff
@@ -10,9 +10,10 @@
 from sympy.matrices.expressions.matexpr import (MatrixExpr, ShapeError,
         Identity, ZeroMatrix)
 from sympy.matrices.matrices import MatrixBase
+from sympy.core.operations import AssocOp
 
 
-class MatMul(MatrixExpr):
+class MatMul(MatrixExpr, AssocOp):
```

Comment:
Regarding `AssocOp`, I added that because the error:

"AttributeError: type object 'MatMul' has no attribute '_from_args'"

was triggered. `Mul` gets its `_from_args` from `AssocOp` hence I added it here. I can look into a another fix for this if this is overkill.

Regarding the "MatMul has two identities", please inform me about what I'm missing, but a `MatrixSymbol` only has the `Identity`, which is (as I understand it) a symbolic representation of the identity matrix (with the integer `1` on the diagonal), which has the property (for any matrix A (This case A is a MatrixSymbol and I is the Identity of )):
 `I*A = A*I =A`
hence the same action from left and right.